lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -12,6 +12,10 @@ def initialize(cache_adaptor = nil)
     end
 
     def cache_backend=(cache_adaptor)
+      if cache_adaptor.class.name == 'ActiveSupport::Cache::MemCacheStore'
+        cache_adaptor.extend(MemCacheStoreCAS)
```

Comment:
> Should we wrap the cache adapter rather than mutating it with `extend`?

I thought about that, but we need to access a bunch of private methods like `merged_options`, etc. So it makes the code much more terse.

> Since ideally we would add CAS support to ActiveSupport::Cache::MemCacheStore itself

`ActiveSupport::Cache` has no CAS support, but maybe it should, it could be worth starting a discussion upstream, since other backends such as Redis could support it. However I'm not sure the interface would be the same anyway.

Reply:
> I'm not sure the interface would be the same anyway.

Yeah, the fact that the interface might not be the same is why I'm thinking it being implemented upstream could cause a conflict.

> I thought about that, but we need to access a bunch of private methods like `merged_options`, etc. So it makes the code much more terse.

Good point.  Maybe this is a good use case for a refinement.

Reply:
> Maybe this is a good use case for a refinement.

Oh, good idea, I never think of that.

Reply:
Actually, not sure if a refinement would be very practical, as we'd have to conditionally activate it. But we only know if it's useful much later at runtime. And nothing guarantees than checking for `defined? AS::Cache::MemCacheStore` at load time would be accurate.

Reply:
Oh right, refinements are statically scoped, which doesn't work as well for extending an optional dependency.  Maybe we should just raise if the AS::Cache::MemCacheStore object responds to `cas` or `cas_multi`, then handle that compatibility in this library if/when CAS support is added upstream.

Reply:
Yeah, that's probably the best idea. 

Anyway if `cas` make it to AS::Cache, it's likely one of us who'll PR it so I think we'll know :)